src/logger/custom/trigger.rs

```diff
@@ -0,0 +1,70 @@
+//!
+//! This module contains custom triggers for when to roll over a log file.
+//! Currently there are the following triggers available:
+//!
+//! * [CustomTrigger]
+//!
+
+use log4rs::append::rolling_file::{policy::compound::trigger::Trigger, LogFile};
+use std::sync::{Arc, Mutex};
+
+/// A trigger which rolls the log once it has passed a certain size
+/// or the global static `LOG_FILE_EXISTS` is `true`.
+#[derive(Copy, Clone, Eq, PartialEq, Hash, Debug, Default)]
+pub struct CustomTrigger {
+    /// The size in bytes after which the file should be rolled over.
+    limit: u64,
+}
+
+impl CustomTrigger {
+    /// Returns a new trigger which rolls the log once it has passed the
+    /// specified size in bytes or the global static `LOG_FILE_EXISTS` is `true`.
+    ///
+    /// # Arguments
+    ///
+    /// * `limit`: the size in bytes, which triggers a roll-over if passed.
+    ///
+    /// returns: CustomTrigger
+    ///
+    /// # Examples
+    ///
+    /// * See [logger::default_logger](crate::logger::default_logger), line 139 for a detailed
+    ///   example.
+
+    pub fn new(limit: u64) -> CustomTrigger {
+        CustomTrigger { limit }
+    }
+}
+
+lazy_static!(
+    /// A global, thread-safe variable that tracks, if the logfile exists at program startup.
+    /// It is set, by [default_logger](crate::logger::default_logger) before the configuration
+    /// is constructed. It **should only be unset by [CustomTrigger::trigger]**, when the logfile
+    /// has been rolled over **ONCE** at program startup!
+    pub static ref LOG_FILE_EXISTS: Mutex<bool> = Mutex::new(false);
```

Comment:
I just noticed, why not use an [AtomicBool](https://doc.rust-lang.org/std/sync/atomic/struct.AtomicBool.html)? It should be faster than a Mutex with bool.

Reply:
See 03c7b33

Reply:
> Why did you revert to `Mutex<bool>`?

The first way in the documentation didn't compile, so I changed it in 0e8519fd4b3afbfeeb172282fe43d4fa3b465227